primitive.cabal

```diff
@@ -56,6 +56,9 @@ Library
                , transformers >= 0.5 && < 0.7
                , template-haskell >= 2.11
 
+  if impl(ghc < 9.4)
+    build-depends: data-array-byte >= 0.1 && < 0.2
```

Comment:
I think you should have `<1.0.1` upper bound, because the type is re-exported and thus its instances too.

In case new stuff is added there (or in `base` for that matter), the `primitive` should version these changes too.

I think there should be problems with tight upper bound, as it's not expected that `data-array-byte` interface will change, even additively.